upf/upf_pfcp.c

```diff
@@ -1629,7 +1629,7 @@ build_pfcp_rules (upf_session_t * sx)
 			     pdr->pdi.teid.teid, idx);
       }
 
-    if ((pdr->pdi.fields & F_PDI_SDF_FILTER))
+    if ((pdr->pdi.fields & (F_PDI_SDF_FILTER | F_PDI_APPLICATION_ID)))
```

Comment:
instead of testing for the flags, wouldn't it make more sense to test for `vec_len(pdr->pdi.acl) != 0`

Reply:
Fixed